### ISSUE TITLE
Pulumi errors with AWS's required format for Elasticache cluster version

### ISSUE THREAD

**Issue by followben** [pulumi/pulumi-aws#1459]
I tried to create an Redis cluster like so:

```python
cluster = aws.elasticache.Cluster(
            name,
            engine="redis",
            engine_version="6.0.5",
            node_type="cache.m4.large",
            num_cache_nodes=1,
            port=6379,
            subnet_group_name=subnetGroup.name,
            security_group_ids=[self.securityGroup.id],
        )
```
Which threw the following error:
```
aws:elasticache:Cluster (apinfra):
    error: 1 error occurred:
    	* error creating ElastiCache Cache Cluster: InvalidParameterValue: Specific version selection is not allowed for '6.0.5', please use '6.x'
    	status code: 400, request id: 1068124d-ae27-474c-8a90-eda86ecf29a7
```

After updating the engine_version to "6.x", the cluster created successfully. However thereafter pulumi up fails with the following:

```
% pulumi up                                                                                    
Previewing update (saberastro/dev)

View Live: https://app.pulumi.com/org/project/dev/previews/821241f1-12ea-w0f0-9fdc-712b8d7aa9e8

     Type                        Name         Plan     Info
     pulumi:pulumi:Stack         project-dev           
     └─ aws:elasticache:Cluster  apinfra               1 error
 
Diagnostics:
  aws:elasticache:Cluster (project):
    error: diffing urn:pulumi:dev::apinfra::aws:elasticache/cluster:Cluster::project: Malformed version: 6.x
```

## Expected behavior
Pulumi should handle wildcard versions as requested by AWS.

## Current behavior
Pulumi throws a Malformed version error when trying to compare the stack.

## Steps to reproduce
See above.

## Context (Environment)
We are currently unable to maintain or update our project's infra with Redis.

## Affected feature
aws.elasticache.Cluster

**Comment by followben** [comment]
Appears this could be related to the terraform provider: https://github.com/hashicorp/terraform-provider-aws/pull/18920

**Comment by stack72** [comment]
Hi @followben 

I believe this has been solved by pulumi-aws 4.2.0 

have you been able to upgrade to use this?

Paul

**Comment by stack72** [comment]
Hi @followben 

I am going to close this out as we have ingested the upstream bug fix - please let me know if this is still an issue and I can investigate further

Thanks

Paul
